packages/clients/textLocator/src/utils/coastalGazetteer/responseInterpreter.ts

```diff
@@ -0,0 +1,110 @@
+import { Feature, FeatureCollection } from 'geojson'
+import levenshtein from 'js-levenshtein'
+import { MultiPolygon } from 'ol/geom'
+import { wgs84ProjectionCode } from '../common'
+import { GeometrySearchState } from '../../plugins/GeometrySearch/types'
+import {
+  ResponseGeom,
+  ResponseName,
+  ResponsePayload,
+  ResponseResult,
+} from './types'
+import { geoJson, idPrefixes, wellKnownText } from './common'
+import { getPrimaryName } from './getPrimaryName'
+
+// arbitrary sort based on input – prefer 1. startsWith 2. closer string
+const sorter =
+  (searchPhrase: string, sortKey: string) =>
+  (a: ResponseName | Feature, b: ResponseName | Feature) => {
+    const aStartsWith = a[sortKey].startsWith(searchPhrase)
+    const bStartsWith = b[sortKey].startsWith(searchPhrase)
+
+    return aStartsWith && !bStartsWith
+      ? -1
+      : !aStartsWith && bStartsWith
+      ? 1
+      : levenshtein(a[sortKey], searchPhrase) -
+        levenshtein(b[sortKey], searchPhrase)
+  }
+
+export const getEmptyFeatureCollection =
+  (): GeometrySearchState['featureCollection'] => ({
+    type: 'FeatureCollection',
+    features: [],
+  })
+
+// for now: merge all geometries, independent of their timeframe
+const geoJsonifyAllGeometries = (
+  geoms: ResponseGeom[],
+  epsg: `EPSG:${string}`
+) =>
+  geoJson.writeGeometryObject(
+    geoms.reduce(
+      (multiPolygon, currentGeom) =>
+        (
+          wellKnownText.readGeometry(currentGeom.WKT, {
+            dataProjection: wgs84ProjectionCode,
+            featureProjection: epsg,
+          }) as MultiPolygon
+        )
+          .getPolygons()
+          .reduce((accumulator, currentPolygon) => {
+            accumulator.appendPolygon(currentPolygon)
+            return accumulator
+          }, multiPolygon),
+      new MultiPolygon([])
+    )
+  )
+
+const featurify =
+  (epsg: `EPSG:${string}`, searchPhrase: string | null) =>
+  (
+    feature: ResponseResult
+  ): GeometrySearchState['featureCollection']['features'][number] | null => {
```

Comment:
I don't want to duplicate that type; changing it in one place should change it in all dependant places; i.e. I'm calling SSOT.

Reply:
This will never be used outside the TextLocator and explicitly requires the `GeometrySearchFeatureProperties` type of the TextLocator. Both solutions work and the rest is opinions. Let's leave it be.